make tests reports failures, but the tests succeed when run individually:

```
otethal@quack:~/src/qore/git/qore$ make tests
...
Passed 116 out of 119 tests. 3 tests failed.
Failed tests:
./examples/test/qore/vars/array.qtest
./examples/test/qore/operators/logic.qtest
./examples/test/qlib/SqlUtil/FreetdsSqlUtil.qtest
*************************************
make: *** [tests] Error 3
otethal@quack:~/src/qore/git/qore$ ./examples/test/qore/vars/array.qtest
QUnit Test "Array test" v1.0
Ran 5 test cases, 5 succeeded (88 assertions)
otethal@quack:~/src/qore/git/qore$ ./examples/test/qore/operators/logic.qtest
QUnit Test "LogicTest" v1.0
Ran 4 test cases, 4 succeeded (15 assertions)
```

--- a/examples/test/qore/vars/array.qtest
+++ b/examples/test/qore/vars/array.qtest
@@ -48,7 +48,7 @@ public class ArrayTest inherits QUnit::Test {
         # Return for compatibility with test harness that checks return value.
         set_return_value(main());
     }
-    
+
     testBasics() {
         my (list a, list b, list c, list d);
 
@@ -101,7 +101,7 @@ public class ArrayTest inherits QUnit::Test {
         c = a + b;
         assertEq(5, c[4], "second list list plus operator concatenation");
     }
-    
+
     testBasics2() {
         list lst1 = (3, 2, 4, 1, 6, 5);
         list lst2 = ("one", "two", "three", "four", "five", "six");
@@ -218,14 +218,14 @@ public class ArrayTest inherits QUnit::Test {
         assertEq((21, 2), (extract lst1, 1, 2, 4), "second list extract");
         assertEq((11, 4), lst1, "final list extract");
     }
-    
+
     testElementDereference() {
         string astr = "hello";
         binary bin = binary(astr);
         assertEq("l", astr[2], "string element dereference");
         assertEq(ord("o"), bin[4], "binary byte dereference");
     }
-    
+
     testRange() {
         assertEq((0, 1,), range(1), "range - basic test");
         assertEq((2, 3, 4, 5), range(2, 5), "range - boundaries test");
@@ -235,7 +235,7 @@ public class ArrayTest inherits QUnit::Test {
         assertEq((-10, -5, 0, 5, 10), range(-10, 10, 5), "range - asc test");
         assertEq((10, 5, 0, -5, -10), range(10, -10, 5), "range - descending step test");
     }
-    
+
     testPseudomethods() {
         list pseudoList = (1, 2, 3, 4, 'a');
         assertEq(NT_LIST, pseudoList.typeCode(), "<list>::typeCode");
@@ -249,4 +249,3 @@ public class ArrayTest inherits QUnit::Test {
         assertEq(True, pseudoList.contains(2), "<list>::contains");
     }
 }
-

--- a/lib/Variable.cpp
+++ b/lib/Variable.cpp
@@ -1083,29 +1083,29 @@ void LValueRemoveHelper::doRemove(AbstractQoreNode* lvalue) {
       return;
    }
 
+   if (t == NT_OPERATOR) {
+      const QoreSquareBracketsOperatorNode* op = dynamic_cast<const QoreSquareBracketsOperatorNode*>(lvalue);
+      if (op) {
+         LValueHelper lvhb(op, xsink, true);
+         if (!lvhb)
+            return;
+
+         bool static_assignment = false;
+         QoreValue tmp = lvhb.remove(static_assignment);
+         if (static_assignment)
+            tmp.ref();
+         rv.assignAssumeInitial(tmp);
+         return;
+      }
+   }
+
    // could be any type if in a background expression
    if (t != NT_TREE) {
       rv.assignInitial(lvalue ? lvalue->refSelf() : 0);
       return;
    }
 
-   // can be only a list or object (hash) reference
-
-   // if it's a list reference, see if the reference exists, if so, then remove it
-   const QoreSquareBracketsOperatorNode* op = dynamic_cast<const QoreSquareBracketsOperatorNode*>(lvalue);
-   if (op) {
-      LValueHelper lvhb(op, xsink, true);
-      if (!lvhb)
-	 return;
-
-      bool static_assignment = false;
-      QoreValue tmp = lvhb.remove(static_assignment);
-      if (static_assignment)
-	 tmp.ref();
-      rv.assignAssumeInitial(tmp);
-      return;
-   }
-
+   // can be only a list reference
    // must be a tree
    assert(t == NT_TREE);
    QoreTreeNode* tree = reinterpret_cast<QoreTreeNode*>(lvalue);

